packages/two_dimensional_scrollables/test/table_view/table_test.dart

```diff
@@ -0,0 +1,1135 @@
+            columnBuilder: (_) => span,
+            rowBuilder: (_) => span,
+            columnCount: 1,
+            pinnedRowCount: 2, // asserts
+            rowCount: 1,
+          );
+        },
+        throwsA(
+          isA<AssertionError>().having(
+            (AssertionError error) => error.toString(),
+            'description',
+            contains('rowCount >= pinnedRowCount'),
+          ),
+        ),
+      );
+      expect(tableView, isNull);
+    });
+  });
+
+  group('TableView.list', () {
+    test('creates correct delegate', () {
+      final TableView tableView = TableView.list(
+        rowBuilder: (_) => span,
+        columnBuilder: (_) => span,
+        cells: const <List<Widget>>[
+          <Widget>[cell, cell, cell],
+          <Widget>[cell, cell, cell]
+        ],
+      );
+      final TableCellListDelegate delegate =
+          tableView.delegate as TableCellListDelegate;
+      expect(delegate.pinnedRowCount, 0);
+      expect(delegate.pinnedRowCount, 0);
+      expect(delegate.rowCount, 2);
+      expect(delegate.columnCount, 3);
+      expect(delegate.columnBuilder(0), span);
+      expect(delegate.rowBuilder(0), span);
+      expect(delegate.children[0][0], cell);
+    });
+
+    test('asserts correct counts', () {
+      TableView? tableView;
+      expect(
+        () {
+          tableView = TableView.list(
+            cells: const <List<Widget>>[
+              <Widget>[cell]
+            ],
+            columnBuilder: (_) => span,
+            rowBuilder: (_) => span,
+            pinnedColumnCount: -1, // asserts
+          );
+        },
+        throwsA(
+          isA<AssertionError>().having(
+            (AssertionError error) => error.toString(),
+            'description',
+            contains('pinnedColumnCount >= 0'),
+          ),
+        ),
+      );
+      expect(
+        () {
+          tableView = TableView.list(
+            cells: const <List<Widget>>[
+              <Widget>[cell]
+            ],
+            columnBuilder: (_) => span,
+            rowBuilder: (_) => span,
+            pinnedRowCount: -1, // asserts
+          );
+        },
+        throwsA(
+          isA<AssertionError>().having(
+            (AssertionError error) => error.toString(),
+            'description',
+            contains('pinnedRowCount >= 0'),
+          ),
+        ),
+      );
+      expect(tableView, isNull);
+    });
+  });
+
+  group('RenderTableViewport', () {
+    testWidgets('parent data and table vicinities',
+        (WidgetTester tester) async {
+      final Map<TableVicinity, UniqueKey> childKeys =
+          <TableVicinity, UniqueKey>{};
+      const TableSpan span = TableSpan(extent: FixedTableSpanExtent(200));
+      final TableView tableView = TableView.builder(
+        rowCount: 5,
+        columnCount: 5,
+        columnBuilder: (_) => span,
+        rowBuilder: (_) => span,
+        cellBuilder: (_, TableVicinity vicinity) {
+          childKeys[vicinity] = UniqueKey();
```

Comment:
This strikes me as odd, every time a cell rebuilds it gets a new UniqueKey? That would almost certainly lead to bugs in a real app. Should we only generate a new key if none exists in the childKeys map?

Reply:
I should probably update the framework test I copied this from then. 🤣